voice-client-javascript/tests/admin/private/statuses.test.js

```diff
@@ -0,0 +1,410 @@
+const helpers = require("../../../../test/test-helper");
+
+const statusFunctions = {};
+let environmentFunction;
+let mockGetCurrentEnvironment;
+
+const mockApplications = {
+  fetch: jest.fn(),
+};
+
+const mockKeys = {
+  fetch: jest.fn(),
+};
+
+const mockTwilioClient = {};
+mockTwilioClient.applications = jest.fn(() => mockApplications);
+mockTwilioClient.applications.list = jest.fn(() => Promise.resolve([]));
+mockTwilioClient.keys = jest.fn(() => mockKeys);
+mockTwilioClient.incomingPhoneNumbers = {
+  list: jest.fn(() => [
+    {
+      phoneNumber: "+15551234567",
+      friendlyName: "(555) 123-4567",
+    },
+    {
+      phoneNumber: "+15557654321",
+      friendlyName: "My Business line",
+    },
+  ]),
+};
+mockTwilioClient.outgoingCallerIds = {
+  list: jest.fn(() => [
+    {
+      phoneNumber: "+18001234567",
+      friendlyName: "My Toll Free",
+    },
+  ]),
+};
+
+const CONTEXT = {
+  ACCOUNT_SID: "AC123",
+  AUTH_TOKEN: "0a123",
+  DOMAIN_NAME: "testing-domain-123.com",
+  PATH: "/check-status",
+  getTwilioClient: jest.fn(() => mockTwilioClient),
+};
+
+const origEnv = {};
+
+describe("voice-client-javascript/admin/private/statuses", () => {
+  beforeAll(() => {
+    const runtime = new helpers.MockRuntime();
+    runtime._addAsset(
+      "/admin/shared.js",
+      "../../../assets/admin/shared.private.js"
+    );
+    helpers.setup(CONTEXT, runtime);
+    // Mock out shared
+    mockGetCurrentEnvironment = jest.fn();
+    const mockShared = jest.mock("../../../assets/admin/shared.private", () => {
+      const actualShared = jest.requireActual("../../../assets/admin/shared.private");
+      return {
+        urlForSiblingPage: actualShared.urlForSiblingPage,
+        getCurrentEnvironment: mockGetCurrentEnvironment
+      };
+    });
+    const mod = require("../../../assets/admin/statuses.private");
+    mod.statuses.forEach((fn) => (statusFunctions[fn.name] = fn));
+    environmentFunction = mod.environment;
+
+    for (let key of Object.keys(process.env)) {
+      origEnv[key] = process.env[key];
+    }
+  });
+
+  afterEach(() => {
+    // Reset the process.env
+    for (let key of Object.keys(process.env)) {
+      if (origEnv[key] === undefined) {
+        delete process.env[key];
+      }
+    }
+    for (let key of Object.keys(origEnv)) {
+      process.env[key] = origEnv[key];
+    }
```

Comment:
Could we move these into two functions? Like:
```
envBackup = helpers.backupEnv();
helpers.restoreEnv(envBackup)
```

I feel like this will be required more often.

Reply:
Added!